### PR TITLE
1.2/967 create route groups redo

### PR DESCRIPTION
**Pull request by charlesstrange2**
Finally, managed to get both ends logging in to do thier own thing!

### CODE REVIEW

File: app/Http/Controllers/Store/Auth/LoginController.php [hunk 1]
```diff
@@ -58,12 +58,12 @@ public function login(Request $request)
             'password' => 'required|min:6',
         ]);
         // Attempt to log the user in
-        if (Auth::guard('web')->attempt(
+        if (Auth::guard('store')->attempt(
```
Comment (katjam) [inline app/Http/Controllers/Store/Auth/LoginController.php]: Nice!

Comment (charlesstrange2) [reply app/Http/Controllers/Store/Auth/LoginController.php]: yeah. I missed that. I was being very dumb. for 2 days!

File: app/Http/Middleware/RedirectIfAuthenticated.php [hunk 2]
```diff
@@ -18,10 +18,15 @@ class RedirectIfAuthenticated
     public function handle($request, Closure $next, $guard = null)
     {
         switch ($guard) {
```
Comment (katjam) [inline app/Http/Middleware/RedirectIfAuthenticated.php]: Very cool. It's neat when you know how.

File: app/Policies/Store/RegistrationPolicy.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+use App\CentreUser;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class RegistrationPolicy
+{
+    use HandlesAuthorization;
+
+    // These are permissions that we need to check a user for.
+    public function updateDiary(CentreUser $user)
```
Comment (katjam) [inline app/Policies/Store/RegistrationPolicy.php]: I know I've said before but why different named policies that check same thing?

File: app/Providers/AuthServiceProvider.php [hunk 4]
```diff
@@ -18,8 +19,9 @@ class AuthServiceProvider extends ServiceProvider
      * @var array
      */
     protected $policies = [
-        \App\Trader::class => \App\Policies\Api\TraderPolicy::class,
-        \App\Voucher::class => \App\Policies\Api\VoucherPolicy::class,
+        Trader::class => \App\Policies\Api\TraderPolicy::class,
```
Comment (katjam) [inline app/Providers/AuthServiceProvider.php]: Much easier to read like this.

File: app/Providers/RouteServiceProvider.php [hunk 5]
```diff
@@ -92,9 +96,23 @@ protected function mapServiceRoutes()
...
-        Route::prefix('data')
+        Route::domain('arcv-service.test')
+            ->prefix('data')
             ->middleware(['web', 'isNotProduction'])
             ->namespace($this->namespace . '\Service')
             ->group(base_path('routes/data.php'));
     }
+
+    /**
+     * Define the "service" routes for the application.
```
Comment (katjam) [inline app/Providers/RouteServiceProvider.php]: You mean "store" routes?

File: app/Providers/RouteServiceProvider.php [hunk 6]
```diff
@@ -92,9 +96,23 @@ protected function mapServiceRoutes()
...
+     *
+     * @return void
+     */
+    protected function mapStoreRoutes()
+    {
+        Route::domain('arcv-store.test')
+            ->middleware(['web'])
```
Comment (katjam) [inline app/Providers/RouteServiceProvider.php]: Just checking still middleware web, right?

File: resources/views/service/auth/login.blade.php [hunk 7]
```diff
@@ -60,7 +60,7 @@
                                     Login
                                 </button>
 
-                                <a class="btn btn-link" href="{{ route('password.request') }}">
+                                <a class="btn btn-link" href="{{ route('admin.password.request') }}">
```
Comment (katjam) [inline resources/views/service/auth/login.blade.php]: Service does = admin, right?

File: resources/views/store/partials/navbar.blade.php [hunk 8]
```diff
@@ -1,12 +1,12 @@
 <div class="subnav">
     <ul>
-        @if ( Request::route()->getName() !== 'service.dashboard' )
+        @if ( Request::route()->getName() !== 'store.dashboard' )
```
Comment (katjam) [inline resources/views/store/partials/navbar.blade.php]: store.base was used elsewhere is this one for specifically dashboard?

File: routes/store.php [hunk 9]
```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+|--------------------------------------------------------------------------
+| Web Routes
+|--------------------------------------------------------------------------
+|
+| Here is where you can register web routes for your application. These
```
Comment (katjam) [inline routes/store.php]: Maybe nix the boilerplate comment

File: routes/store.php [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+| routes are loaded by the RouteServiceProvider within a group which
+| contains the "web" middleware group. Now create something great!
+|
+*/
+
+// Admin (Service) Authentication Routes...
```
Comment (katjam) [inline routes/store.php]: So these should be store now, not admin?

Are they using the same controllers or all pathed under store vs admin namespaceing?

File: routes/store.php [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+]);
+
+Route::post('login', 'Auth\LoginController@login');
+
+Route::post('logout', [
+    'as' => 'store.logout',
+    'uses' => 'Auth\LoginController@logout',
+]);
+
+// Admin (Service) Password Reset Routes...
```
Comment (katjam) [inline routes/store.php]: comment wrong user type.

File: routes/store.php [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+Route::post('password/email', 'Auth\ForgotPasswordController@sendResetLinkEmail')
+    ->name('store.password.email')
+;
+Route::get('password/reset/{token}', 'Auth\ResetPasswordController@showResetForm')
+    ->name('store.password.reset')
+;
+Route::post('password/reset', 'Auth\ResetPasswordController@reset');
+
+// Service Dashboard route
+// Default redirect to Service Dashboard
```
Comment (katjam) [inline routes/store.php]: Store dashboard?

File: tests/Unit/Routes/WebRoutesTest.php [hunk 13]
```diff
@@ -8,14 +8,11 @@
 class WebRoutesTest extends TestCase
 {
     /**
-     * Verify content on dashboard.
+     * Null test,
      *
      * @return void
      */
-    public function testLoginRoute()
+    public function nullTest()
```
Comment (katjam) [inline tests/Unit/Routes/WebRoutesTest.php]: nice test!

File: app/Providers/RouteServiceProvider.php (hunk 5 above)
Comment (charlesstrange2) [reply app/Providers/RouteServiceProvider.php]: ah! yes! thanks.

File: app/Providers/RouteServiceProvider.php (hunk 6 above)
Comment (charlesstrange2) [reply app/Providers/RouteServiceProvider.php]: Yes, the web *middleware* was identical between the installs. 
Saw no reason to change it.

File: routes/store.php (hunk 9 above)
Comment (charlesstrange2) [reply routes/store.php]: OK.

File: routes/store.php (hunk 10 above)
Comment (charlesstrange2) [reply routes/store.php]: yep

File: routes/store.php (hunk 11 above)
Comment (charlesstrange2) [reply routes/store.php]: Ta.

File: routes/store.php (hunk 12 above)
Comment (charlesstrange2) [reply routes/store.php]: :heavy_check_mark:

File: tests/Unit/Routes/WebRoutesTest.php (hunk 13 above)
Comment (charlesstrange2) [reply tests/Unit/Routes/WebRoutesTest.php]: May as well remove the file, actually.

File: app/Policies/Store/RegistrationPolicy.php (hunk 3 above)
Comment (charlesstrange2) [reply app/Policies/Store/RegistrationPolicy.php]: At this poin ti'm loathe to change it without good reason.

**Pull request merged**